robottelo/vm.py

```diff
@@ -89,6 +89,33 @@ def __init__(
         if tag:
             self._target_image = tag + self._target_image
 
+    @property
+    def domain(self):
+        if self._domain is None:
+            try:
+                domain = self.provisioning_server.split('.', 1)[1]
+            except IndexError:
+                raise VirtualMachineError(
+                    u"Failed to fetch domain from provisioning server: {0} "
+                    .format(self.provisioning_server))
+        else:
+            domain = self._domain
+        return domain
+
+    @property
+    def hostname(self):
+        if self._hostname:
+            return self._hostname
+        else:
+            return u'{0}.{1}'.format(self._target_image, self.domain)
+
+    @property
+    def target_image(self):
+        if self._hostname:
+            return self._target_image
```

Comment:
@rochacbruno , self._target_image  is already defined at line 88 one line upper if tag in  __init__ function
```python
    self._target_image = target_image or str(id(self))
```
it's perhaps hidden in github , you need to expend that section